vividus-extension-selenium/src/main/java/org/vividus/bdd/monitor/AbstractScreenshotOnFailureMonitor.java

```diff
@@ -98,10 +101,23 @@ public void onAssertionFailure(AssertionFailedEvent event)
 
     protected abstract Optional<Screenshot> takeAssertionFailureScreenshot(String screenshotName);
 
-    private static boolean takeScreenshotOnFailure(Method method)
+    private boolean takeScreenshotOnFailure(Method method)
     {
-        return method != null && (method.isAnnotationPresent(TakeScreenshotOnFailure.class) || method
-                .getDeclaringClass().isAnnotationPresent(TakeScreenshotOnFailure.class));
+        if (method != null && method.getDeclaringClass().isAnnotationPresent(TakeScreenshotOnFailure.class))
+        {
+            String onlyInDebugMode = method.getDeclaringClass().getAnnotation(TakeScreenshotOnFailure.class)
+                                     .onlyInDebugMode();
+            if (onlyInDebugMode.isEmpty())
+            {
+                return true;
+            }
+            else
+            {
+                return isFailureInDebugMode(onlyInDebugMode);
+            }
+        }
+        return method != null && (method.isAnnotationPresent(TakeScreenshotOnFailure.class)
```

Comment:
And what is the strategy when I have debugOnly annotation at the class level and on the other hand I have a method that annotated to take screenshots always?

Reply:
Added the ability to add an annotation with debugMode at method level.
Behavior is as follows now:  If debugOnly annotation presents at the class level and I have a method that annotated to take screenshots, screenshot will be appear